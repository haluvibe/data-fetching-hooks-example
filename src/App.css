/* App styles */

:root {
  --spacer-small: 12px;
  --spacer-large: 24px;
  --input-height: 32px;
  --input-border-radius: 8px;
  --input-border-padding: 8px;
  --input-border-width: 1px;
  --input-border-color: grey;
  --results-list-element-stack: 8px;
}

.spacer--small {
  margin-bottom: var(--spacer-small);
}

.query-input {
  height: var(--input-height);
  padding: var(--input-border-padding);
  border-radius: var(--input-border-radius);
  border-width: var(--input-border-width);
  border-color: var(--input-border-color);
  width: 100%;
}

.results-list__element {
  margin-bottom: var(--results-list-element-stack);
}

@media only screen and (min-width: 768px){
  .spacer--small {
    margin-bottom: var(--spacer-large);
  }
}
