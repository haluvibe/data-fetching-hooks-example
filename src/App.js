import React, { useState } from 'react';
import useHackerNewsApi from './hooks/useHackerNewsApi';
import MainLayout from './layout/MainLayout';
import ResultsList from './components/ResultsList/ResultsList';
import './App.css';

function App() {
  const [query, setQuery] = useState('redux');
  const { data, isLoading, isError } = useHackerNewsApi(query);

  return (
    <MainLayout>
      <input
        className="query-input"
        type="text"
        value={query}
        onChange={event => setQuery(event.target.value)}
      />
      <div className="spacer--small" />
      <ResultsList query={query} hits={data.hits} isLoading={isLoading} isError={isError} />
    </MainLayout>
  );
}

export default App;
