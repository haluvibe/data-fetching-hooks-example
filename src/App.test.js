import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot();
  });
  it('should handle query input onChange correctly', () => {
    const wrapper = shallow(<App />);
    const queryInput = wrapper.find('.query-input');
    queryInput.instance().value = 'test';
    queryInput.simulate('change');
    expect(wrapper.state().query).toEqual('test');
    expect(wrapper).toMatchSnapshot();
  });
});
