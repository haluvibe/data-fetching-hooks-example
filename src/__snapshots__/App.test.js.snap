// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<MainLayout>
  <input
    className="query-input"
    name="query"
    onChange={[Function]}
    type="text"
    value="redux"
  />
  <div
    className="spacer--small"
  />
  <ResultsList
    hits={
      Array [
        Object {},
      ]
    }
    isError={false}
    isLoading={true}
    query="redux"
  />
</MainLayout>
`;
